pika/compat/__init__.py

```diff
@@ -167,51 +167,35 @@ def get_linux_version(release_str):
 _LOCALHOST = '127.0.0.1'
 _LOCALHOST_V6 = '::1'
 
-if not hasattr(socket, 'socketpair'):
-    # Origin: https://gist.github.com/4325783, by Geert Jansen.  Public domain.
-    def socketpair(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=0):
-        if family == socket.AF_INET:
-            host = _LOCALHOST
-        elif family == socket.AF_INET6:
-            host = _LOCALHOST_V6
-        else:
-            raise ValueError(
-                'Only AF_INET and AF_INET6 socket address families '
-                'are supported')
-        if type != socket.SOCK_STREAM:
-            raise ValueError('Only SOCK_STREAM socket type is supported')
-        if proto != 0:
-            raise ValueError('Only protocol zero is supported')
-
-        # We create a connected TCP socket. Note the trick with
-        # setblocking(False) that prevents us from having to create a thread.
-        lsock = socket.socket(family, type, proto)
+def socketpair(family=socket.AF_INET, type=socket.SOCK_STREAM, proto=0):
```

Comment:
Add a function comment that explains the reasons for having this method - Windows doesn't provide `socket.socketpair`.